presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -287,19 +270,25 @@ private TableHandle getTableHandle(ShowStats node, QualifiedName table)
                     .collect(toImmutableList());
         }
 
-        private List<Expression> buildStatisticsRows(TableStatistics tableStatistics, Map<ColumnHandle, String> columnNames, Map<ColumnHandle, Type> columnTypes)
+        private List<Expression> buildStatisticsRows(TableHandle tableHandle, TableStatistics tableStatistics, Map<String, ColumnHandle> columns)
```

Comment:
Shouldn't we use a `List` to preserve the order of columns as defined in a table?

Reply:
We rely that the Map is ordered. Unfortunately this is something we rely on in the other parts of the code.

Reply:
@arhimondr Not sure I understand this completely. A more robust approach could be to fetch column list from `ConnectorTableMetadata#columns`.